repos/spack_repo/builtin/packages/intel_oneapi_compilers/oneapi_versions.py

```diff
@@ -0,0 +1,473 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import platform
+from typing import List, Optional
+
+from spack.package import *
+
+
+class oneAPIVersions:
+    versions: List["oneAPIVersion"] = []
+
+    @staticmethod
+    def generate_versions():
+        for v in filter(
+            lambda x: x.platform == platform.system().lower(), oneAPIVersions.versions
+        ):
+            version(v.version, expand=False, url=v.cxx.url, sha256=v.cxx.sha)
+            if v.fortran:
+                resource(
+                    name="fortran-installer",
+                    placement="fortran-installer",
+                    when=f"@{v.version}",
+                    expand=False,
+                    url=v.fortran.url,
+                    sha256=v.fortran.sha,
+                )
+            if v.nvidia:
+                resource(
+                    name="nvidia-plugin-installer",
+                    placement="nvidia-plugin-installer",
+                    when=f"@{v.version}",
+                    expand=False,
+                    url=v.nvidia.url,
+                    sha256=v.nvidia.sha,
+                )
+            if v.amd:
+                resource(
+                    name="amd-plugin-installer",
+                    placement="amd-plugin-installer",
+                    when=f"@{v.version}",
+                    expand=False,
+                    url=v.amd.url,
+                    sha256=v.amd.sha,
+                )
+
+
+class oneAPIInstaller:
```

Comment:
I think `InstallResource` might be clearer than `oneAPIInstaller` (I think `oneapi` can be dropped because other packages won't `import *` from this). A short docstring would also be helpful (e.g. "For oneapi resource locations that are not complete, prepends common resource location")

Reply:
I favored the more specific `oneAPIInstaller` so future readers of this package recipe don't see the generic name `InstallerResource` and presume that we have some type of generic mechanic to handle something like this. 

Docstring being added.